### PR TITLE
Fix/missing valid annotations

### PR DESCRIPTION
**Pull request by eschrewe**
I noticed that some of our dto and entity classes are missing an @Valid annotation. This annotation is necessary, because without it, the validator will not inspect the nested objects. 

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/StockDto.java [hunk 1]
```diff
@@ -46,6 +47,7 @@ public abstract class StockDto implements Serializable {
 
     private UUID uuid;
 
+    @Valid
```
Comment (tom-rm-meyer-ISST) [inline backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/StockDto.java:50]: When using the frontend to update a material stock the following error is thrown:

```
jakarta.validation.UnexpectedTypeException: HV000030: No validator could be found for constraint 'jakarta.validation.constraints.Pattern' validating type 'java.lang.Boolean'. Check configuration for 'material.materialFlag'
```

I checked the MaterialDto. For some reason we added a string annotation to the boolean material - which I think is by accident. After removing it, my freehand testing in Frontend + integration test works :)

Please fix.

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Great finding! Faced one issue in testing, please check

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/StockDto.java (hunk 1 above)
Comment (eschrewe) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/StockDto.java:50]: Done, the error was caused by a string pattern matcher that was not suitable for a boolean attribute.

**Review by tom-rm-meyer-ISST — APPROVED** [review]
Works, thanks a lot!

**Pull request merged**